test/Constraints/variadic_generic_functions.swift

```diff
@@ -1,4 +1,4 @@
-// RUN: %target-typecheck-verify-swift
+// RUN: %target-typecheck-verify-swift -disable-availability-checking
```

Comment:
Why is `-disable-availability-checking` necessary now?

Reply:
The struct case emit the following error on my end:
```
/Users/lizhen/Projects/swift-projects/swift/test/Constraints/variadic_generic_functions.swift:107:10: error: unexpected note produced: add @available attribute to enclosing generic struct
  struct Foo<each T> {
         ^
/Users/lizhen/Projects/swift-projects/swift/test/Constraints/variadic_generic_functions.swift:107:14: error: unexpected error produced: parameter packs in generic types are only available in macOS 14.0.0 or newer
  struct Foo<each T> {
             ^
```

Reply:
I removed that flag and placed `struct Foo<each T>` in `test/Constraint/variadic_generic_types.swift` now.